src/main/java/seedu/mtracker/instrument/subinstrument/Forex.java

```diff
@@ -1,15 +1,34 @@
 package seedu.mtracker.instrument.subinstrument;
 
 import seedu.mtracker.instrument.Instrument;
+import seedu.mtracker.ui.TextUi;
 
 public class Forex extends Instrument {
+    protected double entry;
+    protected double exit;
+    protected String expiry;
+    protected String remarks;
 
-    public Forex(String name, double currentPrice, String sentiment) {
+    private static final String FX_ICON = "F";
+
+    public Forex(
+            String name,
+            double currentPrice,
+            String sentiment,
+            double entryPrice,
+            double exitPrice,
+            String deadline,
+            String remark
+    ) {
         super(name, currentPrice, sentiment);
+        this.entry = entryPrice;
+        this.exit = exitPrice;
+        this.expiry = deadline;
+        this.remarks = remark;

```

Comment:
```suggestion
        this.entryPrice = entryPrice;
        this.exitPrice = exitPrice;
        this.expiry = expiry;
        this.remark = remark;

```

To keep naming consistent